lib/ansible/modules/remote_management/manageiq/manageiq_alert_policies.py

```diff
@@ -0,0 +1,337 @@
+
+
+# TODO: I copied this from yaacov's ansible/ansible PR 31233. It should be removed
+# from here once that PR is merged.
+def find_collection_resource_or_fail(module, manageiq, collection_name, **params):
+    """ Searches the collection resource by the collection name and the param passed.
+
+    Returns:
+        the resource as an object if it exists in manageiq, Fail otherwise.
+    """
+    resource = manageiq.find_collection_resource_by(collection_name, **params)
+    if resource:
+        return resource
+    else:
+        msg = "{collection_name} where {params} does not exist in manageiq".format(
+            collection_name=collection_name, params=str(params))
+    module.fail_json(msg=msg)
+
+
+class ManageIQAlerts(object):
+    """ Object to execute alert management operations in manageiq.
+    """
+
+    def __init__(self, manageiq):
+        self.manageiq = manageiq
+
+        self.module = self.manageiq.module
+        self.api_url = self.manageiq.api_url
+        self.client = self.manageiq.client
+        self.url = '{api_url}/alert_definitions'.format(api_url=self.api_url)
+
+    def get_alerts(self):
+        """ Get all alerts from ManageIQ
+        """
+        try:
+            response = self.client.get(self.url + '?expand=resources')
+        except Exception as e:
+            self.module.fail_json(msg="Failed to query alerts: {error}".format(error=e))
+        return response.get('resources', [])
+
+    def validate_hash_expression(self, expression):
+        """ Validate a 'hash expression' alert definition
+        """
+        # hash expressions must have the following fields
+        for key in ['options', 'eval_method', 'mode']:
+            if key not in expression:
+                self.module.fail_json(msg="Hash expression is missing required field {key}".format(key=key))
+
+    def create_policy_dict(self, params):
+        """ Create a dict representing a policy
+        """
+        if params['expression_type'] == 'hash':
+            # hash expression supports depends on https://github.com/ManageIQ/manageiq-api/pull/76
+            self.validate_hash_expression(params['expression'])
+            expression_type = 'hash_expression'
+        else:
+            expression_type = 'miq_expression'
+
+        # build the policy
+        policy = dict(description=params['description'],
+                      db=params['entity_type'],
+                      options=params['options'],
+                      enabled=params['enabled'])
+
+        # add the actual expression.
+        policy.update({expression_type: params['expression']})
+
+        return policy
+
+    def add_alert_policy(self, policy):
+        """ Add a new alert policy to ManageIQ
+        """
+        try:
+            result = self.client.post(self.url, action='create', resource=policy)
+            msg = "Alert policy {description} created successfully: {details}".format(description=policy['description'], details=result)
```

Comment:
the long msg lines are not nice, try to find a way to shorten them.

for example:
```
msg_template = "Alert policy {description} created {status}: {details}"
msg_data = dict(description=policy['description'], status='successfully', details=result)

msg = msg_template.format(msg_data)
```